### PR TITLE
Introduce EditorConfig

### PR DESCRIPTION
**Pull request by szepeviktor**
- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [ ] util/update-modules has been ran
- [ ] Documentation updated
- [ ] Test suite update

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
```
Comment (svetasmirnova) [inline .editorconfig:5]: We have indent size = 3 for Perl code

Comment (szepeviktor) [reply .editorconfig:5]: 😲

Recalculating ...

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+indent_size = unset
```
Comment (szepeviktor) [inline .editorconfig]: Oh no! This line was left in the configuration file.

File: .editorconfig (hunk 1 above)
Comment (szepeviktor) [reply .editorconfig:5]: What is the first batch I should reindent?
Should I leave artistic indentations in place?
```perl
   my ($package, undef, $line) = caller 0;
   @_ = map { (my $temp = $_) =~ s/\n/\n# /g; $temp; }
        map { defined $_ ? $_ : 'undef' }
        @_;
```

**Review by EvgeniyPatlan — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (svetasmirnova) [reply .editorconfig:5]: Yes, please, leave them.

**Comment by svetasmirnova** [comment]
@szepeviktor, what is the status of your PR? Are you still working on it and want to keep it draft?

**Comment by szepeviktor** [comment]
Finishing it right away.

**Comment by szepeviktor** [comment]
Using consistent (by simply counting spaces) indentation is a no-go for perl code.
I've just installed cpanminus and issued `cpanm Perl::Tidy`

Parenthesis spacing is so chaotic in the code that I'm unable to select a style.
https://perltidy.sourceforge.net/tutorial.html#bt--pt--sbt:-Container-tightness
Should it be -pt=0 or -pt=1 or -pt=2? Please help me.

For example
```
perltidy -b -i=3 -l=120 -pt=1 bin/pt-visual-explain
git diff -- bin/pt-visual-explain
```

**Comment by svetasmirnova** [comment]
Percona Toolkit is based on Maatikit product. Here are the coding standards for Maatkit: https://code.google.com/archive/p/maatkit/wikis/CodingStandards.wiki. They were not fully followed in recent years but I believe we can restore them.

Here is example config for perltidy:
Example .perltidyrc file: 
```
-l=78 # Max line width is 78 cols 
-i=3 # Indent level is 4 cols 
-ci=3 # Continuation indent is 4 cols 
-b # In-place update
#-st # Output to STDOUT 
-se # Errors to STDERR 
-vt=2 # Maximal vertical tightness 
-cti=0 # No extra indentation for closing brackets 
-pt=1 # Medium parenthesis tightness 
-bt=1 # Medium brace tightness 
-sbt=1 # Medium square bracket tightness 
-bbt=1 # Medium block brace tightness 
-nsfs # No space before semicolons 
-nolq # Don't outdent long quoted strings 
-wbb="% + - * / x != == >= <= =~ !~ < > | & >= < = **= += *= &= <<= &&= -= /= |= >>= ||= .= %= ^= x=" 
```

**Comment by szepeviktor** [comment]
Thank you.
I will run perltidy on all perl files and that is the end of this PR.

```
-b
-l=100 # Max line width
-i=3 # Indent level
-ci=3 # Continuation indent
-se # Errors to STDERR
-vt=2 # Maximal vertical tightness
-cti=0 # No extra indentation for closing brackets
-pt=1 # Medium parenthesis tightness
-bt=1 # Medium brace tightness
-sbt=1 # Medium square bracket tightness
-bbt=1 # Medium block brace tightness
-nsfs # No space before semicolons
-nolq # Don't outdent long quoted strings
-wbb="% + - * / x != == >= <= =~ !~ < > | & >= < = **= += *= &= <<= &&= -= /= |= >>= ||= .= %= ^= x="
```

**Comment by szepeviktor** [comment]
```
git ls-files -- 'lib/**.pm' | xargs -t -L 1 -- perltidy
```

perltidy makes artistic indentations go away
```diff
- sub _d {
-    my ($package, undef, $line) = caller 0;
-    @_ = map { (my $temp = $_) =~ s/\n/\n# /g; $temp; }
-         map { defined $_ ? $_ : 'undef' }
-         @_;
-    print STDERR "# $package:$line $PID ", join(' ', @_), "\n";
- }

+    sub _d {
+       my ( $package, undef, $line ) = caller 0;
+       @_ = map { ( my $temp = $_ ) =~ s/\n/\n# /g; $temp; }
+          map { defined $_ ? $_ : 'undef' } @_;
+       print STDERR "# $package:$line $PID ", join( ' ', @_ ), "\n";
+    }
```

**Comment by szepeviktor** [comment]
I have modified line length and added `-b` to overwrite files ([in the above comment](https://github.com/percona/percona-toolkit/pull/681#issuecomment-1792705762)).

@svetasmirnova This is above my pay grade.
What do you think?

**Comment by svetasmirnova** [comment]
@szepeviktor I think it does not make sense to format code just in sake to make it compatible with perltidy. Let's have only 3 spaces in the config and don't do other formatting. We can always add additional configuration rules into .editorconfig later.

**Comment by szepeviktor** [comment]
Done.

**Review by svetasmirnova — APPROVED** [review]

**Pull request merged**